src/compute/src/render/top_k.rs

```diff
@@ -472,18 +512,23 @@ where
     // NOTE(vmarcos): The arranged input operator name below is used in the tuning advice
     // built-in view mz_internal.mz_expected_group_size_advice.
     input
-        .mz_arrange::<KeyValSpine<(Row, u64), _, _, _>>("Arranged TopK input")
-        .mz_reduce_abelian::<_, KeyValSpine<_, _, _, _>>("Reduced TopK input", {
-            move |key, source, target: &mut Vec<(R, Diff)>| {
+        .mz_arrange::<RowRowSpine<_, _>>("Arranged TopK input")
+        .mz_reduce_abelian::<_, Tr>("Reduced TopK input", {
+            move |mut hash_key, source, target: &mut Vec<(Tr::ValOwned, Diff)>| {
+                let binding = SharedRow::get();
+                let mut row_builder = binding.borrow_mut();
+                let mut row_packer = row_builder.packer();
```

Comment:
Can we move obtaining the `row_packer` closer to where it's actually used? I think we can move it into the error-handling block and just above the loop that produces the output. For me, this would make it easier to understand the scope where it's used.

Reply:
Absolutely, I moved obtaining the `row_builder` down a bit and just called `packer()` on it directly where necessary right before cloning.